src/main/java/com/javastreets/mulefd/MuleXmlElement.java

```diff
@@ -102,47 +110,60 @@ public static List<MuleComponent> fillComponents(Element flowElement,
+    final Optional<ComponentItem> item = findByName(knownComponents, keyName);
+
+    return item.isPresent() ? Optional.of(createKnownComponent(element, item.get())) : Optional.empty();
+  }
+
+  private static MuleComponent createKnownComponent(Element element, ComponentItem item) throws XPathExpressionException {
+    final String[] wildcards = element.getTagName().split(":");
+    String name = wildcards != null && wildcards.length > 1
+            ? wildcards[1]
+            : item.getOperation();
+    final MuleComponent mc = new MuleComponent(item.getPrefix(), name);
+    mc.setSource(item.isSource());
+    if (!item.getConfigAttributeName().trim().isEmpty()) {
+      mc.setConfigRef(Attribute.with(item.getConfigAttributeName(),
+          element.getAttribute(item.getConfigAttributeName())));
     }
 
-    if (item != null) {
-      String name = item.getOperation();
-      if (wildcards != null && wildcards.length > 1) {
-        name = wildcards[1];
-      }
-      KnownMuleComponent mc = new KnownMuleComponent(item.getPrefix(), name);
-      mc.setSource(item.isSource());
-      if (!item.getConfigAttributeName().trim().isEmpty()) {
-        mc.setConfigRef(Attribute.with(item.getConfigAttributeName(),
-            element.getAttribute(item.getConfigAttributeName())));
-      }
-      if (!item.getPathAttributeName().trim().isEmpty()) {
-        mc.setPath(Attribute.with(item.getPathAttributeName(),
-            element.getAttribute(item.getPathAttributeName())));
-      }
-      mc.setAsync(item.isAsync());
-      muleComponentList.add(mc);
+    final String pathExpression = item.getPathAttributeName().trim();
+    if (!pathExpression.isEmpty()) {
+      final String evaluatedPath = isXpath(pathExpression)
+              ? evaluateXpathOnElement(element, pathExpression)
+              : element.getAttribute(item.getPathAttributeName());
+      mc.setPath(Attribute.with(item.getPathAttributeName(), evaluatedPath));
     }
+    mc.setAsync(item.isAsync());
+    return mc;
+  }
 
+  private static Optional<ComponentItem> findByName(final Map<String, ComponentItem> components, final String keyName) {
+    final Optional<ComponentItem> foundItemByKey = Optional.ofNullable(components.get(keyName));
+    if (foundItemByKey.isPresent())
+      return foundItemByKey;
+    if (keyName.contains(":")) {
+      final String wildcardEntry = keyName.split(":")[0] + ":*";
+      return Optional.ofNullable(components.get(wildcardEntry));
+    }
+    return Optional.empty();
+  }
+
+  private static boolean isXpath(final String expression) {
+    return expression.startsWith(XPATH_INDICATOR);
+  }
 
+  private static String evaluateXpathOnElement(final Element element, final String expression) throws XPathExpressionException {
+    XPath xPath = XPathFactory.newInstance().newXPath();
+    return (String)xPath.compile(expression.substring(XPATH_INDICATOR.length())).evaluate(element, XPathConstants.STRING);
```

Comment:
Does `XPathExpressionException` must bubble up? Can it be wrapped and thrown as an instance of `com.javastreets.mulefd.drawings.DrawingException` (unchecked exception)?

Reply:
Sure, can be done this way. Is probably nicer as it does not spoil the code as much.

Reply:
Resolved in https://github.com/manikmagar/mulefd/pull/300/commits/96008f40d4cd1db59539a3c7a9bd7fc529314f8d